File: android/app/src/main/java/com/expensetracker/app/domain/model/Transaction.kt
package com.expensetracker.app.domain.model

import java.util.Calendar
import java.util.TimeZone

data class Transaction(
    val id: Long = 0,
    val kind: TransactionKind,
    val amount: Double,
    val categoryId: Long? = null,
    val accountId: Long? = null,
    val toAccountId: Long? = null,
    val date: Long,
    val note: String = "",
    val tagIds: List<Long> = emptyList(),
    val attachments: List<String> = emptyList(),
    val needsReview: Boolean = false,
    val source: String = "",
    val affectsAsset: Boolean = true,
    val isDeleted: Boolean = false,
    val refundOf: Long? = null,
    val createdAt: Long = System.currentTimeMillis()
) {
    val monthKey: String
        get() {
            val calendar = Calendar.getInstance(TimeZone.getDefault())
            calendar.timeInMillis = date
            val year = calendar.get(Calendar.YEAR)
            val month = calendar.get(Calendar.MONTH) + 1
            return String.format("%04d-%02d", year, month)
        }
}

File: ExpenseTracker/Models/Transaction.swift
import Foundation
import SwiftData

enum TransactionKind: Int, Codable {
    case expense = 0
    case income = 1
    case transfer = 2
    case refund = 3

    var title: String {
        switch self {
        case .expense: "支出"
        case .income: "收入"
        case .transfer: "转账"
        case .refund: "退款"
        }
    }
}

@Model
final class Transaction {
    @Attribute(.unique) var id: UUID
    var kindRaw: Int
    var amount: Double
    var categoryID: UUID?
    var accountID: UUID?
    var toAccountID: UUID?
    var date: Date
    var note: String
    var tagIDs: [UUID]
    var attachmentPaths: [String]
    var createdAt: Date
    var updatedAt: Date
    var needsReview: Bool
    var source: String
    var affectsAsset: Bool
    var fromRecurring: Bool
    var refundOfID: UUID?
    var isDeleted: Bool

    /// 去重 hash（来源+商户+金额+日期）
    var dedupHash: String?
    /// 对账用: 记账后账户快照值
    var balanceSnapshot: Double?

    var kind: TransactionKind {
        get { TransactionKind(rawValue: kindRaw) ?? .expense }
        set { kindRaw = newValue.rawValue }
    }

    var monthKey: String {
        let f = DateFormatter()
        f.dateFormat = "yyyy年M月"
        return f.string(from: date)
    }

    init(
        id: UUID = UUID(),
        kind: TransactionKind = .expense,
        amount: Double = 0,
        categoryID: UUID? = nil,
        accountID: UUID? = nil,
        toAccountID: UUID? = nil,
        date: Date = Date(),
        note: String = "",
        tagIDs: [UUID] = [],
        attachmentPaths: [String] = [],
        needsReview: Bool = false,
        source: String = "manual",
        affectsAsset: Bool = true,
        fromRecurring: Bool = false,
        refundOfID: UUID? = nil,
        dedupHash: String? = nil
    ) {
        self.id = id
        self.kindRaw = kind.rawValue
        self.amount = amount
        self.categoryID = categoryID
        self.accountID = accountID
        self.toAccountID = toAccountID
        self.date = date
        self.note = note
        self.tagIDs = tagIDs
        self.attachmentPaths = attachmentPaths
        self.createdAt = Date()
        self.updatedAt = Date()
        self.needsReview = needsReview
        self.source = source
        self.affectsAsset = affectsAsset
        self.fromRecurring = fromRecurring
        self.refundOfID = refundOfID
        self.isDeleted = false
        self.dedupHash = dedupHash
    }
}
